packages/expo-speech/src/Speech/ExponentSpeech.web.ts

```diff
@@ -32,9 +49,19 @@ export default {
       message.volume = options.volume;
     }
     if ('_voiceIndex' in options && options._voiceIndex != null) {
-      const voices = window.speechSynthesis.getVoices();
+      const voices = await getVoices();
       message.voice = voices[Math.min(voices.length - 1, Math.max(0, options._voiceIndex))];
     }
+    if (typeof options.voice === 'string') {
+      const voices = await getVoices();
+      message.voice =
+        voices[
+          Math.max(
+            0,
+            voices.findIndex((voice) => voice.voiceURI === options.voice)
+          )
+        ];
+    }
```

Comment:
If the voice provided in the options don't match any available voice we fallback to the first one in the list.
Not doing any clever stuff with `voice.default` here, as `_voiceIndex` does the same